### PR TITLE
fix(dropdownMenu): Fix positioning issues

### PR DESCRIPTION
**Pull request by vuluongj20**
Rather than using `react-aria`'s `useOverlayPosition`, we can use our own `useOverlay` util, which does a better job at overflow prevention.

**Before:**
Screen Shot 2022-10-07 at 2 20 12 PM

**After:**
Screen Shot 2022-10-07 at 2 20 52 PM

### CODE REVIEW

File: static/app/components/actions/ignore.tsx [hunk 1]
```diff
@@ -236,9 +236,8 @@ const IgnoreActions = ({
       </IgnoreButton>
       <DropdownMenuControl
         size="sm"
-        trigger={({props: triggerProps, ref: triggerRef}) => (
+        trigger={triggerProps => (
           <DropdownTrigger
-            ref={triggerRef}
```
Comment (vuluongj20) [inline static/app/components/actions/ignore.tsx:241]: The `triggerProps` from `useOverlay` contains a `ref` prop, so we don't need the second argument.

File: static/app/components/dropdownMenu.tsx [hunk 2]
```diff
@@ -188,9 +149,8 @@ function Menu({
         items={node.value.children as MenuItemProps[]}
         trigger={trigger}
         menuTitle={node.value.submenuTitle}
-        placement="right top"
+        position="right-start"
```
Comment (vuluongj20) [inline static/app/components/dropdownMenu.tsx:152]: just like with `<Tooltip />`

File: static/app/components/dropdownMenuItem.tsx [hunk 3]
```diff
@@ -199,7 +194,14 @@ const MenuItem = ({
 
   return (
     <MenuListItem
-      ref={ref}
+      ref={el => {
```
Comment (malwilley) [inline static/app/components/dropdownMenuItem.tsx:197]: It looks like there is a `utils/mergeRefs.tsx` that you can use here

Comment (vuluongj20) [reply static/app/components/dropdownMenuItem.tsx]: Oh woah that's neat! Updated in https://github.com/getsentry/sentry/pull/39800/commits/5920587235261208eaf10bd6d2b511ccd3145fe3

**Review by malwilley — APPROVED** [review]
Looks good!

**Pull request merged**